src/app/georeferencer/qgsgeoreftoolmovepoint.cpp

```diff
@@ -39,10 +32,36 @@ bool QgsGeorefToolMovePoint::isCanvas( QgsMapCanvas *canvas )
 
 void QgsGeorefToolMovePoint::canvasMoveEvent( QgsMapMouseEvent *e )
 {
-  emit pointMoved( e->pos() );
+  QgsPointLocator::Match match;
+  if ( !mStartPointMapCoords.isEmpty() )
+  {
+    const QgsPointXY pnt = toMapCoordinates( e->pos() );
+    match = canvas()->snappingUtils()->snapToMap( pnt );
+    mSnapIndicator->setMatch( match );
+  }
+  emit pointMoving( match.isValid() ? match.point() : toMapCoordinates( e->pos() ) );
 }
 
 void QgsGeorefToolMovePoint::canvasReleaseEvent( QgsMapMouseEvent *e )
 {
-  emit pointReleased( e->pos() );
+  if ( e->button() & Qt::LeftButton )
+  {
+    if ( mStartPointMapCoords.isEmpty() )
+    {
+      emit pointBeginMove( mStartPointMapCoords );
+    }
+    else
+    {
+      mSnapIndicator->setMatch( QgsPointLocator::Match() );
+      emit pointEndMove( toMapCoordinates( e->pos() ) );
+    }
+  }
+  else if ( e->button() & Qt::RightButton )
+  {
+    if ( !mStartPointMapCoords.isEmpty() )
+    {
+      mSnapIndicator->setMatch( QgsPointLocator::Match() );
+      emit pointCancelMove( toMapCoordinates( e->pos() ) );
```

Comment:
Can you also add a `keyPressEvent()` override to handle cancelling using `Esc` too?

Reply:
@uclaros , implemented